weave/index/index_pandas.py

```diff
@@ -337,6 +341,11 @@ def get_parents(self, basket_address, **kwargs):
         data = data.drop_duplicates(
             subset=['uuid', 'generation_level']
         ).reset_index(drop=True)
+
+        # Keep all of the rows where the generation_level is less than or
```

Comment:
Enforcing the max generation level here means we still have to do all the computation to get the entire list of descendants even if we only want the first few generations.
Would it be better to check gen_level <= max_gen_level in the earlier for loop to avoid the unnecessary computation?

Reply:
I agree. The idea of having max levels is to avoid doing the full compute chain. Fortunately you have gen_level as a kwarg you can check against at the start of the function.